src/main/java/cinema/service/impl/CustomUserDetailService.java

```diff
@@ -0,0 +1,25 @@
+package cinema.service.impl;
+
+import cinema.model.User;
+import cinema.security.UserDetailsFactory;
+import cinema.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailService implements UserDetailsService {
+    @Autowired
+    private UserService userService;
```

Comment:
```suggestion
    private final UserService userService;
```
it's nice to keep the code consistent